torch/csrc/jit/passes/freeze_module.h

```diff
@@ -0,0 +1,23 @@
+/** \brief This file defines passes to freeze pytorch module..
+ *
+ * The passes have python-bindings and can be invoked directly or as a part of
+ * general optimization pipeline (details TBD).
+ */
+#pragma once
+
+#include <torch/csrc/jit/ir.h>
+#include <torch/csrc/jit/script/module.h>
+
+/** \brief Freeze Module, i.e., Assume all atrribute are constants.
+ *
+ * constant propagate attribute and optimize module attributes agressively
```

Comment:
Would be worth mentioning that we're doing all of this on a clone of the original module.